lib/std/c.zig

```diff
@@ -1702,17 +1704,44 @@ pub const S = switch (native_os) {
     .linux => linux.S,
     .emscripten => emscripten.S,
     .wasi => struct {
-        pub const IEXEC = @compileError("TODO audit this");
+        // Match wasi-libc's libc-bottom-half/headers/public/__mode_t.h
+        // NOTE: this S struct is also used by Wasi-without-libc
```

Comment:
No good. wasi without libc should never reach into std.c.

Reply:
Okay, wasn't sure how pure you wanted to the wasi-without-libc path to be.  I can fork `Stat`, `timespec` and `S` in std.posix.

Reply:
All the wasi-without-libc structs are fully defined in posix.zig now.  I added `.S`, `.timespec` and `.Stat`, plus a couple integers (`nlink_t`, etc).  This should just be the minimum necessary to make the wasi-without-libc code in posix.zig work.  I based them on the wasi-with-libc structs, and pruned obviously unsupported bits and unnecessary padding.